src/adagrams.js

```diff
@@ -1,8 +1,81 @@
-const Adagrams = {
-  drawLetters() {
-    // Implement this method for wave 1
-  },
+class Adagrams {
+  constructor() {
+  }
+  static drawLetters() {
+    const letterDist = {
+      A: 9, B: 2, C: 2, D: 4, E: 12, F: 2, G: 3, H: 2, I: 9, J: 1, K: 1, L: 4, M: 2, N: 6,O: 8, P: 2, Q: 1, R: 6, S: 4, T: 6,U: 4, V: 2, W: 2, X: 1, Y: 2, Z: 1
+    };
+    let letterPool = []
+    // create letterPool
+    for (const letter in letterDist) {
+      for (let i = 0; i < letterDist[letter]; i++) {
+        letterPool.push(letter);
+      } 
+    }
+    // shuffle letterPool, used the Fisher Yates model from https://bost.ocks.org/mike/shuffle/
+    let letterPoolClone = letterPool.map((x) => x);
+
+    for(let i = letterPool.length; i > 0; i--){
+      let x = Math.floor(Math.random() * i);
+      let temp = letterPoolClone[i];
+      letterPoolClone[i] = letterPoolClone[x];
+      letterPoolClone[x] = temp
+    }
+  const chosenLetters = letterPoolClone.slice(0,10);
+  return chosenLetters;
+  }
+
+  static usesAvailableLetters(input, lettersInHand) {
+    for(let i = 0; i < input.length; i++) {
+      if(lettersInHand.includes(input[i])){
+        lettersInHand.splice(lettersInHand.indexOf(input[i]), 1 );
+      } else {
+        return false;
+      }
+    }
+  return true
+  }
+
+  static scoreWord(word) {
+    let numPoints = 0
+    let letterScores = {
+    'A':1, 'E':1, 'I':1, 'O':1, 'U':1, 'L':1, 'N':1, 'R':1, 'S':1, 'T':1, 'D':2, 'G':2, 'B':3, 'C':3, 'M':3, 'P':3, 'F':4, 'H':4, 'V':4, 'W':4, 'Y':4, 'K':5,'J':8, 'X':8,'Q':10, 'Z':10
+    }
+
+    for(let i = 0; i < word.length; i++){
+      numPoints += letterScores[word[i].toUpperCase()];
+    }
```

Comment:
In general I suggest using `forEach` or `for-in` loops instead of the oldstyle `for` just because it's easy to make a typo given those three different sections.

```suggestion
    for(const letter of word)
      numPoints += letterScores[letter.toUpperCase()];
    }
```